### ISSUE TITLE
isolated declarations: default-initialized any parameter loses `| undefined`

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#25290]
### Description

Oxc’s isolated declarations transform drops `| undefined` from a parameter with a default initializer.

This was caught by our ID ecosystem CI.

### Reproduction

```ts
export function ssrGetDynamicModelProps(
  existingProps: any = {},
  model: unknown,
): { checked: true } | { value: any } | null {
  return {}
}
```

### Expected output

```ts
export declare function ssrGetDynamicModelProps(existingProps: any | undefined, model: unknown): {
    checked: true;
} | {
    value: any;
} | null;
```

### Actual output

```ts
export declare function ssrGetDynamicModelProps(existingProps: any, model: unknown): {
    checked: true;
} | {
    value: any;
} | null;
```

The default initializer means the parameter may be omitted, so the generated declaration should preserve `| undefined`.

### Regression information

- [Most recent successful ecosystem CI job](https://github.com/oxc-project/monitor-oxc/actions/runs/30886098735/job/91917944598)
- [First failing ecosystem CI job](https://github.com/oxc-project/monitor-oxc/actions/runs/30898504486/job/91957480520)
- [Vue commit exposing the issue](https://github.com/vuejs/core/commit/37bf752bf1ff3df023f38fe43a8f12564db23b85)
